src/packet/tcp.rs.in

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2014, 2015 Robert Clipsham <robert@octarineparrot.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use packet::Packet;
+use packet::PrimitiveValues;
+use packet::ip::IpNextHeaderProtocol;
+
+use pnet_macros_support::types::*;
+
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+/// Represents a TCP Packet
+#[packet]
+pub struct Tcp {
+    source: u16be,
+    destination: u16be,
+    seq: u32be,
```

Comment:
For libpnet, I'd prefer to have descriptive names rather than posix-style abbreviations, eg. "sequence" rather than "seq", "acknowledgement" rather than "ack".